File: Algorithms/python/bfs.py
import numpy as np
import os

# READING MAP
# =============================================================================

def read_map_file(filename):
    with open(filename, 'r') as file:
        lines = file.readlines()

    # Remove newlines from each line and convert them into integers
    map_data = [[int(cell) for cell in line.strip()] for line in lines]

    return np.array(map_data)

# Replace 'map.txt' with the path to your map file
map_file = os.getcwd() + '\Algorithms\student-pack-stuff\matlab\map_1.txt'
map_array = read_map_file(map_file)

# Print the resulting 2D array
print(map_array)
# READING MAP
# =============================================================================

# Method 1: Using nested lists
# rows = 15
# columns = 19

# # Initializing the array with zeros
# array = [[0 for _ in range(columns)] for _ in range(rows)]

# # Print the array
# for row in array:
#     print(f'{row}, ')



def GetMapFromFile(filepath = '\student-pack-stuff\matlab\map_1.txt'):
    # ASSUMPTIONS: Map is always 15 (rows) x 19 (columns) and ends with a new line


    # 1 == Path, 0 == Cannot got

    # get map txt from file path
    # read map file and convert into 2d array
    # return 2d array

    return 

File: Algorithms/c/bfs.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_ROWS 100 // Maximum number of rows in the map
#define MAX_COLS 100 // Maximum number of columns in the map
int test = 0;

int main() {
    int map[MAX_ROWS][MAX_COLS];
    int rows = 0;

    // Open the file
    FILE *file = fopen("map.txt", "r");
    if (file == NULL) {
        printf("Failed to open the file.\n");
        return 1;
    }

    // Read lines and convert characters to integers
    char line[MAX_COLS];
    while (fgets(line, sizeof(line), file) != NULL) {
        for (int i = 0; line[i] != '\0' && line[i] != '\n'; i++) {
            map[rows][i] = line[i] - '0';
        }
        rows++;
    }

    // Close the file
    fclose(file);

    // Print the resulting 2D array
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < MAX_COLS; j++) {
            if (map[i][j] == 0) {
                printf("0 ");
            } else if (map[i][j] == 1) {
                printf("1 ");
            }
        }
        printf("\n");
    }

    return 0;
}